module/os/linux/zfs/zpl_ctldir.c

```diff
@@ -398,6 +403,9 @@ zpl_snapdir_getattr_impl(const struct path *path, struct kstat *stat,
 	(void) request_mask, (void) query_flags;
 	struct inode *ip = path->dentry->d_inode;
 	zfsvfs_t *zfsvfs = ITOZSB(ip);
+	dsl_dataset_t *ds = NULL;
```

Comment:
Lately we prefer declaring variables near or at where they are first used, to limit their syntactic and visual scope.